plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -493,57 +541,84 @@ protected function initialize()
+                    // set to false in order to avoid automatically creating a
+                    // type for the index that was just created
+                    $index->create(
+                        $this->config['index']['configuration'],
+                        ['recreate' => true]
+                    );
                 }
 
-                $this->index->create(
-                    $this->config['index']['configuration'],
-                    ['recreate' => true]
-                );
-            }
+                // Iterate over types (actor, informationobject, ...)
+                foreach ($this->mappings as $indexName => $indexProperties) {
+                    $indexName = 'Qubit'.sfInflector::camelize($indexName);
 
-            // Load and normalize mappings
-            $this->loadAndNormalizeMappings();
+                    if ($indexType != $indexName) {
+                        continue;
+                    }
 
-            // Iterate over types (actor, informationobject, ...)
-            foreach ($this->mappings as $typeName => $typeProperties) {
-                $typeName = 'Qubit'.sfInflector::camelize($typeName);
+                    // Define mapping in elasticsearch
+                    $mapping = new \Elastica\Type\Mapping();
 
-                // Define mapping in elasticsearch
-                $mapping = new \Elastica\Type\Mapping();
-                $mapping->setType($this->index->getType($typeName));
-                $mapping->setProperties($typeProperties['properties']);
+                    // Setting a dummy type since it is required in ES 6.x
+                    // but it can be removed in 7.x when it becomes optional
+                    $mapping->setType($index->getType(self::ES_TYPE));
+                    $mapping->setProperties($indexProperties['properties']);
 
-                // Parse other parameters
-                unset($typeProperties['properties']);
-                foreach ($typeProperties as $key => $value) {
-                    $mapping->setParam($key, $value);
-                }
+                    // Parse other parameters
+                    unset($this->mapping[$indexName]->indexProperties['properties']);
+                    foreach ($this->mapping[$indexName]->indexProperties as $key => $value) {
+                        $mapping->setParam($key, $value);
+                    }
 
-                $this->log(sprintf('Defining mapping %s...', $typeName));
-                $mapping->send();
+                    $this->log(sprintf('Defining mapping %s...', $indexName));
+
+                    // In ES 7.x this should be changed to:
+                    // $mapping->send($index, [ 'include_type_name' => false ])
+                    // which can be removed in 8.x since that is the default behaviour
+                    // and will have be removed by 9.x when it is discontinued
+                    $mapping->send();
+                }
             }
```

Comment:
@jraddaoui it does feel odd to define the mappings outside the `if`. I let it be because I wasn't sure if there was some historical reason or if there was a scenario where the index existed but could not be opened because it didn't have mappings declared. In my testing changing it to:
```
try {                                                             
  $index->open();                                      
} catch (\Elastica\Exception\ResponseException $e) {
// If the index has not been initialized, create it
```
didn't seem to cause any problems (both in the scenario when the index existed and when it didn't). Let me know if you want me to make this change

Reply:
Yeah, changing the catch and removing the if makes sense to me.